### ISSUE TITLE
Request: Have auto uninstall compare names with spaces removed

### ISSUE THREAD

**Issue by teknowledgist** [chocolatey/choco#1245]
### What You Are Seeing?
Automatic uninstall won't work for some packages if the application display name has spaces but the package name is made by concatenating the words.  

### What would be nice to see:
The Get-UninstallRegistryKey would remove spaces from the `Display Name` as pulled from the uninstall registry keys as well as removing spaces from the provided `SoftwareName` (i.e. the `PackageName` for the automatic uninstall) before comparing them.

There is a (slight) possibility of automatic uninstall failing due to two applications installed with the same string of letters differing only by the placement of spaces.  This could be avoided by re-comparing (including spaces) the `DisplayName` and `SoftwareName` if multiple (un)install registry keys are found.

### What would happen?
Packages with a packagename created by joining words could automatically uninstall without a maintainer-created `chocolateyuninstall.ps1` file.

##### Comment
I am not aware of any existing packages with this "problem", so it's obviously not urgent.  However, I am creating a package (bluebeamvu) that does require a `chocolateyuninstall.ps1` script only for this very reason.  It should be a pretty easy fix too; I'm just not able to submit PRs for this repository.

**Comment by ferventcoder** [comment]
@teknowledgist I'm not sure that this is accurate. Auto Uninstaller doesn't use the package name at all to determine what it is installing, only a capture of registry uninstaller changes from a before and after snapshot.

**Comment by teknowledgist** [comment]
Hrmmm.  It's been so long, I forget exactly what I was seeing.  :smile: 

I believe the autouninstall was failing when I was testing the blubeamvu package because it wasn't finding the "bluebeamvu" registry key and was believing the application was removed through other means.  When I created a chocolateyuninstall.ps1 that included `Get-UninstallRegistryKey -SoftwareName "Bluebeam Vu*"` it worked.  

Maybe something else was going on, but the autouninstaller calling `Get-UninstallRegistryKey` with the package name fit what I was seeing.
